util/platform.zig

```diff
@@ -8,11 +8,37 @@ const mac = @import("os/osx.zig");
 const win = @import("os/windows.zig");
 
 pub const OsInfo = struct {
+    allocator: std.mem.Allocator,
     platform: []const u8,
     cpu: []const u8,
     cpu_cores: u32,
     memory_total: []const u8,
     // ... other system information
+
+    pub fn deinit(self: OsInfo) void {
+        self.allocator.free(self.cpu);
```

Comment:
When running the basic example i get an allocation error. It indicates a mismatch in the size of memory allocated versus the size of memory attempted to be freed. 

I tried it running on a macos x86_64:
`[gpa] (err): Allocation size 40 bytes does not match free size 39`

Reply:
Thanks, I don't get that on my Debian machine so tricky to debug from here. I might see about running the examples as part of CI.

Reply:
`[]const u8` should be a stack pointer; interesting that this is causing an allocation error. For the Linux code, I only use the allocator to dupe the string, to avoid a pointer to temporary memory, so I guess this explains why @bens doesn't get this issue on Linux. I didn't touch the OSX code since I don't have a Mac around for testing; my guess is that the call to `exec` leaves some heap memory that wants to be freed.

Reply:
Checked out the system-info branch with the fixed alloc 👍 this seems to resolve the error. What do you think how we should proceed? Bring this one into this PR, or as a follow up.

Reply:
The only down side I know of for that branch is that I had to use 4KB of stack for the Windows version to not run out of space, which seems like quite a bit. It could use static memory for scratch instead, it's just that doesn't get freed afterwards, or could use different amounts for each platform. I don't know much about Windows development but from what I could find processes are [allocated 1MB by default](https://learn.microsoft.com/en-us/windows/win32/procthread/thread-stack-size), so 4KB might be fine there... It's not the sort of code that's likely to be executed in a recursion context, and after the first call it doesn't need stack again.

Reply:
Can bring it in to this PR, it's nice to not have to debug the issue at all :smile:

Reply:
That would be cool 👍